### PR TITLE
api/cmp_service

### PR DESCRIPTION
**Pull request by alhiwatan**
Add support for CMP service.  

- #160

### CODE REVIEW

File: applovin_max/ios/Classes/AppLovinMAX.m [hunk 1]
```diff
@@ -1299,7 +1332,20 @@ - (void)logInvalidAdFormat:(MAAdFormat *)adFormat
...
 {
-    [self log: @"ERROR: Failed to execute %@() - please ensure the AppLovin Flutter Native module has been initialized by calling 'AppLovinMAX.initialize(...);'!", callingMethod];
+    [self logUninitializedAccessError: callingMethod withResult: nil];
+}
+
+- (void)logUninitializedAccessError:(NSString *)callingMethod withResult:(nullable FlutterResult)result
+{
+    NSString *message = [NSString stringWithFormat:@"ERROR: Failed to execute %@() - please ensure the AppLovin MAX React Native module has been initialized by calling 'AppLovinMAX.initialize(...);'!", callingMethod];
+
+    if (!result)
```
Comment (thomasmso) [inline applovin_max/ios/Classes/AppLovinMAX.m:1342]:
```suggestion
    if ( !result )
```

**Review by thomasmso — APPROVED** [review]

File: applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAX.java [hunk 2]
```diff
@@ -557,6 +559,38 @@ public void setConsentFlowDebugUserGeography(final String userGeography)
...
+    {
+        if ( sdk == null )
+        {
+            logUninitializedAccessError( "showCmpForExistingUser", result );
+            return;
+        }
+
+        AppLovinCmpService cmpService = sdk.getCmpService();
+        cmpService.showCmpForExistingUser( getCurrentActivity(), (@Nullable final AppLovinCmpError error) -> {
+            if ( error == null )
```
Comment (santoshbagadi) [inline applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAX.java:572]:
```suggestion
        cmpService.showCmpForExistingUser( getCurrentActivity(), (@Nullable final AppLovinCmpError error) -> {

            if ( error == null )
```

**Review by santoshbagadi — APPROVED** [review]

**Comment by Vollinator** [comment]
When will it be merged and released?

File: applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAX.java [hunk 3]
```diff
@@ -557,6 +559,38 @@ public void setConsentFlowDebugUserGeography(final String userGeography)
...
+
+            result.success( error.getCmpCode() );
+        } );
+    }
+
+    public void hasSupportedCmp(final Result result)
+    {
+        if ( sdk == null )
+        {
+            logUninitializedAccessError( "showCmpForExistingUser", result );
```
Comment (ritamsarmah) [inline applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAX.java:586]:
```suggestion
            logUninitializedAccessError( "hasSupportedCmp", result );
```

**Comment by thomasmso** [comment]
@Vollinator - Likely EoW

**Pull request merged**